src/psygnal/_signal.py

```diff
@@ -1280,49 +1290,50 @@ def __call__(self, args: tuple[object, ...]) -> bool:
     def __eq__(self, other: object) -> bool:
         return (
             isinstance(other, _BoundMethodCaller)
-            and self._ref == other._ref
-            and self._method_name == other._method_name
+            and self._obj_ref == other._obj_ref
+            and self._func_ref == other._func_ref
         )
 
+    def _method(self) -> MethodType | None:
+        obj = self._obj_ref()
+        func = self._func_ref()
+        if obj is None or func is None:
+            return None
+        return self._method_type(func, obj)
```

Comment:
Why need to rebind instead of directly calling `func(obj, *args)`? (requires bigger refactor of course)

Reply:
ah, good idea.  lemme try that as well.  (`weakref.WeakMethod` obviously rebinds to return the "original" object... but you're right that we don't need that here)

Reply:
that helped!  no benchmark regression 🎉

Reply:
sidenote: I recently got my beta invite to https://codspeed.io ... will add here soon